pkg/cloudproviders/aws/controllers/interruption/controller.go

```diff
@@ -0,0 +1,60 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package interruption
+
+import (
+	"context"
+
+	"k8s.io/utils/clock"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	awscache "github.com/aws/karpenter/pkg/cloudproviders/aws/cache"
+	"github.com/aws/karpenter/pkg/cloudproviders/aws/controllers/providers"
+	"github.com/aws/karpenter/pkg/cloudproviders/aws/events"
+	"github.com/aws/karpenter/pkg/controllers/state"
+)
+
+// Controller is an AWS interruption controller.
+type Controller struct {
+	kubeClient                client.Client
+	clk                       clock.Clock
+	cluster                   *state.Cluster
+	recorder                  *events.Recorder
+	provider                  *providers.SQS
+	unavailableOfferingsCache *awscache.UnavailableOfferings
+}
+
+func NewController(kubeClient client.Client, clk clock.Clock, recorder *events.Recorder, cluster *state.Cluster,
+	sqsProvider *providers.SQS, unavailableOfferingsCache *awscache.UnavailableOfferings) *Controller {
+
+	return &Controller{
+		kubeClient:                kubeClient,
+		clk:                       clk,
+		cluster:                   cluster,
+		recorder:                  recorder,
+		provider:                  sqsProvider,
+		unavailableOfferingsCache: unavailableOfferingsCache,
+	}
+}
+
+func (c *Controller) Reconcile(_ context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	return reconcile.Result{}, nil
```

Comment:
If we're not implementing it just yet, can we just include this in the NTH PR?

Reply:
This is a staging PR to ready the core logic. This is intended to break up the PR so the NTH PR isn't so large

Reply:
It was also intended to show the act of the controller construction and registration flow looks like